src/library/scala/Function.scala

```diff
@@ -28,6 +28,15 @@ object Function {
   /** The constant function */
   def const[T, U](x: T)(y: U): T = x
 
+  /**
+   * Returns a function that always returns its input argument as result.
+   * Which could be used as a function argument passed to `map` or `flatMap` methods.
+   * Learn more from [[https://en.wikipedia.org/wiki/Identity_function Identity_function]]
+   * @tparam T the type of the input and output value of the function
+   * @return a function that always returns its input argument as result.
+   */
+  def identity[T]: T => T = t => t
```

Comment:
As is, this will allocate a new instance for each invocation, IMO this should return a cached instance.

Reply:
it was constIdentity.asInstanceOf[T=>T] ,but I  changed that before pr.

This thing is what I learnt from you:stuck_out_tongue_winking_eye:

Reply:
@viktorklang like this:

https://github.com/hepin1989/scala/commit/c29ac874d18dd1f8f81e81cb74aab8b8ac1680a9#diff-cdead8ee05a4284557bc258dc86e300dR39